### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by skumarya-tibco** [kubernetes/kubernetes#119316]
### What happened?

When running an application on kind version 0.20.0 with k8s version 1.27.3 the application is failing with the error "panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1398670]"
whereas when we are trying with kind version 0.18.0 and k8s version 1.27.3 the application is working smoothly.
Also, we tried explicitly passing kind node image version 1.26.6 then it works smoothly but it fails when the default kind node image version is 1.27.3 it shows the above error.

### What did you expect to happen?

when explicitly wee provide node image version of k8s 1.26.6 the application works smoothly for 1.27.3 too it should.

### How can we reproduce it (as minimally and precisely as possible)?

create a kind cluster with version 0.20.0 which will by default pull node image version 1.27.3 of Kubernetes when application is deployed we see this panic error
 

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>
1.27.3

### Cloud provider

<details>

</details>
local kind

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>
kind version 0.20.0

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by brianpursley** [comment]
Strange that it seems to be the specific combination of kind 0.20.0 with k8s 1.27.3.

It also must have something to do with the application you are running, because when I just try to run nginx, I don't have any problem.  Admittedly, that is an overly-simplistic test and doesn't mean this isn't an issue.

Do you get a call stack with the panic error?

Have you tried it with kind 0.19.0?  I'm wondering if we can zero in on exactly when the problem started happening.

**Comment by skumarya-tibco** [comment]
have tried 
kind version: kind v0.19.0 go1.20.4 darwin/amd64
k8s node image: kindest/node:v1.27.1 byDefault it pulls this version of k8s.
where I see the same error as kind version 0.20.0 with k8s version
error:
panic: runtime error: invalid memory address or nil pointer dereference[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1398670]goroutine 1 [running]:k8s.io/client-go/discovery.convertAPIResource(...)	k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:88k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc0006b9548, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})	k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:69 +0x5f0k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc0006b8978, 0x15}, {0xc00097a140, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})	k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:35 +0x2f8k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x2000107?)	k8s.io/client-go@v0.26.0/discovery/discovery_client.go:310 +0x478k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0x40d485?)	k8s.io/client-go@v0.26.0/discovery/discovery_client.go:198 +0x5ck8s.io/client-go/discovery/cached/memory.(*memCacheClient).refreshLocked(0xc00005ccc0)	k8s.io/client-go@v0.26.0/discovery/cached/memory/memcache.go:222 +0x57k8s.io/client-go/discovery/cached/memory.(*memCacheClient).GroupsAndMaybeResources(0xc00005ccc0)	k8s.io/client-go@v0.26.0/discovery/cached/memory/memcache.go:128 +0xc5k8s.io/client-go/discovery/cached/memory.(*memCacheClient).ServerGroups(0xc000988f00?)	k8s.io/client-go@v0.26.0/discovery/cached/memory/memcache.go:148 +0x19k8s.io/client-go/discovery/cached/disk.(*CachedDiscoveryClient).ServerGroups(0xc000988f00)	k8s.io/client-go@v0.26.0/discovery/cached/disk/cached_discovery.go:118 +0x172k8s.io/client-go/discovery.ServerGroupsAndResources({0x21acb40, 0xc000988f00})	k8s.io/client-go@v0.26.0/discovery/discovery_client.go:397 +0xc8k8s.io/client-go/discovery/cached/disk.(*CachedDiscoveryClient).ServerGroupsAndResources(0x0?)	k8s.io/client-go@v0.26.0/discovery/cached/disk/cached_discovery.go:101 +0x25helm.sh/helm/v3/pkg/action.GetVersionSet({0x7f1594867b30?, 0xc000988f00?})	helm.sh/helm/v3/pkg/action/action.go:312 +0x42helm.sh/helm/v3/pkg/action.(*Configuration).getCapabilities(0xc0001b3880)	helm.sh/helm/v3/pkg/action/action.go:258 +0x97helm.sh/helm/v3/pkg/action.(*Install).RunWithContext(0xc0004f51e0, {0x21a5218?, 0xc000898e40}, 0x2?, 0xc00091fb20?)	helm.sh/helm/v3/pkg/action/install.go:238 +0x60dmain.runInstall({0xc0006be370?, 0x2, 0xb}, 0xc0004f51e0, 0xc00091fca8?, {0x218ca60?, 0xc0001ae008})	helm.sh/helm/v3/cmd/helm/install.go:279 +0x86emain.newUpgradeCmd.func2(0xc0005b8300?, {0xc0006be370?, 0x2, 0xb})	helm.sh/helm/v3/cmd/helm/upgrade.go:123 +0x476github.com/spf13/cobra.(*Command).execute(0xc0005b8300, {0xc0006be2c0, 0xb, 0xb})	github.com/spf13/cobra@v1.6.1/command.go:916 +0x862github.com/spf13/cobra.(*Command).ExecuteC(0xc0006aef00)	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bcgithub.com/spf13/cobra.(*Command).Execute(...)	github.com/spf13/cobra@v1.6.1/command.go:968main.main()	helm.sh/helm/v3/cmd/helm/helm.go:83 +0x235

whereas, when explicitly changing the kind version 0.19.0 node image version image: kindest/node:v1.26.6 then the application works smoothly. 
kind version 0.18.0 by default pulls the image (kindest/node:v1.26.3) where the application works fine.

**Comment by brianpursley** [comment]
Based on the call stack, I suspect you are running into this problem https://github.com/kubernetes/kubernetes/issues/118361#issuecomment-1579198407

The call stack is showing client-go version 0.26.0, but there was a problem in client-go versions 0.26.0-0.26.3 related to aggregated discovery which was promoted to beta (enabled by default) in k8s version 1.27.  The comment I linked says the problem was fixed in client-go version 0.26.4

The calls stack you provided also shows that this is coming from helm.  Can you try using helm version 3.12.2?  https://github.com/helm/helm/releases/tag/v3.12.2

**Comment by skumarya-tibco** [comment]
yes when updating the helm the panic error is gone and the deployments are working smooth.
Thank you.
